test/stack.js

```diff
@@ -55,6 +55,10 @@ describe('Stack', function () {
           return stack.length() === 2
         })
       })
+
+      it('return what is popped', function() {
+        expect(stack.pop()).to.deep.equal('bar')
```

Comment:
Why do you need `deep` here?  It should just be equal to the string?  (I call this out in case you're returning the wrong structure from `pop`.)